omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

```diff
@@ -1537,15 +1537,15 @@ def get_zoom_level_scale(self, image, region, max_width):
         # e.g. {0: 1.0, 1: 0.25, 2: 0.0625, 3: 0.03123, 4: 0.01440}
         # Pick zoom such that returned image is below MAX size
         max_level = len(zm_levels.keys()) - 1
-
-        # Maximum size that the rendering engine will render without OOM
-        max_plane = self.conn.getDownloadAsMaxSizeSetting()
+        # Maximum size that the rendering engine will render
+        max_sizes = self.conn.getMaxPlaneSize()
 
         # start big, and go until we reach target size
         zm = 0
         while (zm < max_level and
                zm_levels[zm] * width > max_width or
-               zm_levels[zm] * width * zm_levels[zm] * height > max_plane):
+               zm_levels[zm] * width > max_sizes[0] or
+               zm_levels[zm] * height > max_sizes[1]):
```

Comment:
By comparing width and height separately you get a more conservative (smaller) image, E.g. if your image width is 3 x the height, then the plane size you get will be 1/3 of the big image cut-off.

Alternatively you could leave this logic unchanged and do `max_plane = max_sizes[0] * max_sizes[1]` above?

Reply:
Fixed in [7d25b9f](https://github.com/ome/omero-figure/pull/553/commits/7d25b9fb49959ba7f143c70927573efb62c9667e). This is consistent with the way plane sizes are used in `is_big_image` above https://github.com/ome/omero-figure/blob/87d179a0db8b6ef3cf9b81f3fe0c4879e263fb55/omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py#L1526-L1529 
but also server-side to indicate the requirement of pyramidal levels

https://github.com/ome/omero-romio/blob/1d30fafc4e06c5511cfbb24c25a753925ffb2eb4/src/main/java/ome/io/nio/PixelsService.java#L686